bash/enforce_all_nagios_check/enforce_all_nagios_check.sh

```diff
@@ -1,77 +1,155 @@
 #!/bin/bash
 ##-------------------------------------------------------------------
 ## File : enforce_all_nagios_check.sh
-## Author : Denny <denny@dennyzhang.com>
+## Author : Denny <denny.zhang001@gmail.com>
 ## Description :
 ## --
 ## Created : <2015-06-24>
-## Updated: Time-stamp: <2016-04-09 09:35:43>
+## Updated: Time-stamp: <2016-04-18 15:58:19>
 ##-------------------------------------------------------------------
-skip_check_pattern=${1:-""}
-ignore_check_warn=${2:-"0"}
-nagios_check_dir=${3:-""}
 
-if [ -z "$nagios_check_dir" ]; then
-    server_name=`hostname`
-    nagios_check_dir="/etc/nagios3/conf.d/$server_name"
-fi
-
-if [ ! -d $nagios_check_dir ]; then
-    echo "ERROR: $nagios_check_dir doesn't exist"
-    exit 1
-fi
-cd $nagios_check_dir
-
-failed_checks=""
-skipped_checks=""
-IFS=$'\n'
-for f in `ls -1 *.cfg`; do
-    if grep '^ *host_name *' $f 2>/dev/null 1>/dev/null; then
-        host_name=$(grep '^ *host_name *' $f | awk -F' ' '{print $2}' | head -n 1)
-        for check in `grep '^ *check_command' $f | awk -F' ' "{print $2}" | awk -F'!' '{print $2}'`; do
-            command="/usr/lib/nagios/plugins/check_nrpe -H $host_name -c $check"
-            if [ -n "$skip_check_pattern" ]; then
-                if echo $check | grep -iE "$skip_check_pattern" 2>/dev/null 1>/dev/null; then
-                    echo "skip check: $command"
-                    skipped_checks="${skipped_checks}${check};"
-                    continue
+function check_one_server(){
+    local nagios_check_dir=${1}
+    local skip_check_pattern=${2}
+    cd $nagios_check_dir
+    shift
+    shift
+    
```

Comment:
I don't understand you, could you please have a example?

nagios_check_dir, skip_check_pattern is already local.